kafka-client-examples/e2e-test/src/main/java/dev/responsive/examples/regression/ResultsComparatorService.java

```diff
@@ -118,13 +127,13 @@ protected void run() throws Exception {
             baseline.remove();
             matches++;
           } else if (responsive.hasNext() && baseline.hasNext()) {
-            LOG.error(
-                "ANTITHESIS NEVER: Expected to consume records in order, but they were not"
-                    + "in order. Most recent record from responsive is {} and most recent up"
-                    + "record from baseline is {}",
+            Assert.unreachable(String.format(
+                "Expected to see identical output records in identical order, but the next set "
```

Comment:
@agavra is this a fair re-wording of this error message? It's not just about checking ordering, it's about correctness, right?

Also, it seems like it would be possible for the test to "pass" when we actually never see any output from one side at all. I did add the `Assert#sometimes` cases above to help so now it should enforce that we see output from each side at least once, but in theory the Responsive side could stall after the first record and we wouldn't catch this right?

Reply:
correct and correct 👍 I hadn't added the "sometimes" assertion since the SDK wasn't available but that was my intention.

Reply:
Do you think we should try to assert that a certain number of records minimum are processed? Or that one side doesn't lag for too long? Not quite sure what a good trigger would be but we want to make sure some kind of alarm is raised if one side only ever processes a single record and then gets stuck for the rest of time.

Might not be a good way to enforce this since the Responsive side can lag the Streams side, especially with chaos testing, and I'm not sure we can put a bound on either. Did you have any thoughts about this when you wrote the test?

Reply:
if I understand antithesis correctly, it will continuously run the test until it doesn't really notice any new scenarios being hit so it's just important that the setup ran. from that point on code coverage and randomness will handle the rest.

cc @rodesai